--- Transactions.js	2015-04-16 13:18:31.000000000 -0700
+++ Transactions.js	2015-04-16 13:18:52.000000000 -0700
@@ -264,10 +264,10 @@ InviteClientTransaction.prototype.cancel
 
   this.cancel += 'Content-Length: 0\r\n\r\n';
 
-  // Send only if a provisional response (>100) has been received.
-  if(this.state === C.STATUS_PROCEEDING) {
+  // // Send only if a provisional response (>100) has been received.
+  // if(this.state === C.STATUS_PROCEEDING) {
     this.transport.send(this.cancel);
-  }
+  // }
 };
 
 InviteClientTransaction.prototype.receiveResponse = function(response) {
