--- Grammar.js	2015-05-11 16:18:21.000000000 -0700
+++ Grammar.js	2015-05-11 16:18:44.000000000 -0700
@@ -182,7 +182,7 @@
           function() {
                               options.data.password = text(); },
           function() {
-                              options.data.host = text().toLowerCase();
+                              options.data.host = text();
                               return options.data.host; },
           function() {
                             options.data.host_type = 'domain';
