--- Hacks.js	2014-11-24 14:22:26.000000000 -0800
+++ Hacks.js	2015-01-22 14:02:29.000000000 -0800
@@ -11,7 +11,7 @@
   AllBrowsers: {
     maskDtls: function (message) {
       if (message.body) {
-        message.body = message.body.replace(/ UDP\/TLS\/RTP\/SAVP/gmi, " RTP/SAVP");
+        // message.body = message.body.replace(/ UDP\/TLS\/RTP\/SAVP/gmi, " RTP/SAVP");
       }
     },
     unmaskDtls: function (sdp) {
@@ -26,7 +26,7 @@
        * Opera now supports DTLS by default as well.
        *
        **/
-      return sdp.replace(/ RTP\/SAVP/gmi, " UDP/TLS/RTP/SAVP");
+      return sdp; // return sdp.replace(/ RTP\/SAVP/gmi, " UDP/TLS/RTP/SAVP");
     }
   },
   Firefox: {
