--- Session.js	2015-05-04 09:11:40.000000000 -0700
+++ Session.js	2015-05-04 09:12:03.000000000 -0700
@@ -665,8 +665,8 @@
         if(this.status === C.STATUS_CONFIRMED) {
           this.logger.log('re-INVITE received');
           // Switch these two lines to try re-INVITEs:
-          //this.receiveReinvite(request);
-          request.reply(488, null, ['Warning: 399 sipjs "Cannot update media description"']);
+          this.receiveReinvite(request);
+          //request.reply(488, null, ['Warning: 399 sipjs "Cannot update media description"']);
         }
         break;
       case SIP.C.INFO:
