--- MediaHandler.js	2015-05-15 11:24:18.000000000 -0700
+++ MediaHandler.js	2015-05-15 11:25:54.000000000 -0700
@@ -464,7 +464,7 @@
       .then(SIP.Utils.promisify(pc, 'setLocalDescription'))
       .then(function onSetLocalDescriptionSuccess() {
         var deferred = SIP.Utils.defer();
-        if (pc.iceGatheringState === 'complete' && (pc.iceConnectionState === 'connected' || pc.iceConnectionState === 'completed')) {
+        if (pc.iceConnectionState === 'complete' || pc.iceConnectionState === 'completed') {
           deferred.resolve();
         } else {
           self.onIceCompleted.promise.then(deferred.resolve);
