--- WebRTC.js	2014-10-16 16:20:50.000000000 -0700
+++ WebRTC2.js	2014-10-16 16:22:19.000000000 -0700
@@ -17,10 +17,20 @@
     return _isSupported;
   }
 
-  WebRTC.MediaStream = environment.MediaStream;
-  WebRTC.getUserMedia = environment.getUserMedia;
-  WebRTC.RTCPeerConnection = environment.RTCPeerConnection;
-  WebRTC.RTCSessionDescription = environment.RTCSessionDescription;
+  if ((typeof window === 'undefined' || typeof window.navigator === 'undefined')
+    && typeof process === 'object')
+  {
+    var wrtc = require('wrtc');
+    WebRTC.MediaStream = wrtc.MediaStream;
+    WebRTC.getUserMedia = wrtc.getUserMedia;
+    WebRTC.RTCPeerConnection = wrtc.RTCPeerConnection;
+    WebRTC.RTCSessionDescription = wrtc.RTCSessionDescription;
+  } else {
+    WebRTC.MediaStream = environment.MediaStream;
+    WebRTC.getUserMedia = environment.getUserMedia;
+    WebRTC.RTCPeerConnection = environment.RTCPeerConnection;
+    WebRTC.RTCSessionDescription = environment.RTCSessionDescription;
+  }
 
   if (WebRTC.RTCPeerConnection && WebRTC.RTCSessionDescription) {
     WebRTC.getUserMedia = SIP.Utils.addPromise(WebRTC.getUserMedia);
